src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts

```diff
@@ -52,9 +53,10 @@ export function useEstimatedGasFee({
 }
 
 export function useSwapEstimatedGasFee(gasSettings: GasSettings | undefined) {
-  const { assetToSell, chainId = ChainId.mainnet, quote } = useSyncedSwapQuoteStore();
-  const { data: estimatedGasLimit, isFetching } = useSwapEstimatedGasLimit({ chainId, assetToSell, quote });
+  const { assetToSell, quote } = useSyncedSwapQuoteStore();
+  const chainId = useSwapsStore(s => s.inputAsset?.chainId || ChainId.mainnet);
 
+  const { data: estimatedGasLimit, isFetching } = useSwapEstimatedGasLimit({ chainId, assetToSell, quote });
```

Comment:
Won't this change cause the gas limit and gas fee to be recomputed more often, and potentially behave erratically, since the `chainId` no longer changes together with the `assetToSell` and `quote`? It's also no longer guaranteed to match the `assetToSell`.

Is there some reason the `chainId` from the store is needed?

Also, can you broadly explain the changes in this PR?

Reply:
we start without a `quote`, so the first `estimatedGasLimit` was always a placeholder for mainnet, even if/when the input token of another chain was selected
if we don't have a quote it won't try to fetch it, just using the right chainId to have a good enough placeholder 

inside `useSwapEstimatedGasLimit` added a check to disable the query when `chainId !== assetToSell.chainId`

inside `useSwapEstimatedGasLimit` returning `{ gasLimit, chainId }` from the `queryFn` makes so we know which chain that gasLimit refers to, and we can disregard it and return a 'placeholder' if the chainId changed but the query didn't update yet

the `gasSettings` arg is updating with the inputAsset chainId, before while we didn't have a new quote it was calculating with `gasLimit` of chain `A` (because of `keepPreviousData`) but using the updated `gasSettings` of chain `B`